### PR TITLE
Converts local filesystem plugin to service providers

### PR DESCRIPTION
**Pull request by laoneo**
### Summary of Changes
Converts local filesystem plugin to service providers.

### Testing Instructions
Open the media manager.

### Actual result BEFORE applying this Pull Request
All works.

### Expected result AFTER applying this Pull Request
All works.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by richard67** [comment]
@laoneo Unit tests are failing: https://ci.joomla.org/joomla/joomla-cms/60934/1/11 . I think it **_could_** be related to this PR.

**Comment by dgrammatiko** [comment]
I have tested this item :white_check_mark: successfully on d080d1e62fd13b92a1afe0e510825933c7774472<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39642">issues.joomla.org/tracker/joomla-cms/39642</a>.</sub>

File: plugins/filesystem/local/src/Extension/Local.php [hunk 1]
```diff
@@ -84,7 +108,7 @@ public function getDisplayName()
     public function getAdapters()
     {
         $adapters = [];
-        $directories = $this->params->get('directories', '[{"directory": "images", "thumbs": 0}]');
+        $directories = $this->params->get('directories', '[{"directory": "images"}]');
```
Comment (dgrammatiko) [inline plugins/filesystem/local/src/Extension/Local.php]:
```suggestion
        $directories = $this->params->get('directories', '[{"directory": "images", "thumbs": 0}]');
```

**Review by dgrammatiko — CHANGES_REQUESTED** [review]

File: plugins/filesystem/local/src/Extension/Local.php [hunk 2]
```diff
@@ -94,19 +118,10 @@ public function getAdapters()
...
-                    $directoryEntity->thumbs = 0;
-                }
-
-                $adapter = new \Joomla\Plugin\Filesystem\Local\Adapter\LocalAdapter(
-                    $directoryPath,
-                    $directoryEntity->directory,
-                    $directoryEntity->thumbs,
-                    [200, 200]
-                );
+                $adapter = new LocalAdapter($directoryPath, $directoryEntity->directory);
```
Comment (dgrammatiko) [inline plugins/filesystem/local/src/Extension/Local.php]:
```suggestion
          if (!isset($directoryEntity->thumbs)) {
               $directoryEntity->thumbs = 0;
           }

           $adapter = new LocalAdapter(
               $directoryPath,
               $directoryEntity->directory,
               $directoryEntity->thumbs,
               [200, 200]
           );
```

File: plugins/filesystem/local/services/provider.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+     * @return  void
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    public function register(Container $container)
+    {
+        $container->set(
+            PluginInterface::class,
+            function (Container $container) {
+                $plugin     = new Local(
```
Comment (Quy) [inline plugins/filesystem/local/services/provider.php]:
```suggestion
                $plugin = new Local(
```

**Comment by dgrammatiko** [comment]
I have tested this item :white_check_mark: successfully on 085fe56197ca8f1db4a47ea15e103d8e31e15e43<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39642">issues.joomla.org/tracker/joomla-cms/39642</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39642">issues.joomla.org/tracker/joomla-cms/39642</a>.</sub>

**Comment by obuisard** [comment]
Thank you Allon @laoneo for the PR.

**Pull request merged**